test/unit/utils.test.js

```diff
@@ -51,4 +51,60 @@ describe('utils.js', () => {
     });
     expect(decrypted).to.deep.equal(VALUE2);
   });
+  describe('parse', () => {
+    it('should parse stringified string', () => {
+      const original = 'some string';
+      const parsed = parse(JSON.stringify(original));
+
+      expect(parsed).to.equal(original);
+    });
+    it('should parse stringified object', () => {
+      const original = {
+        f1: 'single capital E but not a num',
+        f2: {
+          f3: 'nested',
+          f4: [{ f5: 'nested in array' }],
+        },
+      };
+      const parsed = parse(JSON.stringify(original));
+
+      expect(parsed).to.deep.equal(original);
+    });
+    it('should parse stringified array with field value: exponential string', () => {
+      const original = [
+        '33E44',
+      ];
+      const parsed = parse(JSON.stringify(original));
+
+      expect(parsed).to.deep.equal(original);
+    });
+    it('should parse stringified object with field value: exponential string', () => {
+      const original = {
+        f1: '33E44',
+      };
+      const parsed = parse(JSON.stringify(original));
+
+      expect(parsed).to.deep.equal(original);
+    });
+    it('should parse stringified object with field value: exponential number', () => {
+      const original = {
+        f1: 33E44,
+      };
+      const parsed = parse(JSON.stringify(original));
+
+      expect(parsed).to.deep.equal(original);
+    });
+    it('should parse stringified number', () => {
+      const original = 42;
+      const parsed = parse(JSON.stringify(original));
+
+      expect(parsed).to.deep.equal(original);
+    });
+    it('should NOT parse stringified exponential number', () => {
```

Comment:
it should NOT parse a non-stringified string that looks like an exponential number